### PR TITLE
[MIG][11.0] web_widget_datepicker_options: Migrate module to v11

### PR DESCRIPTION
**Pull request by yaniaular**
This migrates the module [web_widget_datepicker_options](https://github.com/OCA/web/tree/10.0/web_widget_datepicker_options) which had not been migrated yet from v10.

This applies some changes to the module, to conform new guidelines and
conventions for version 11.0.

In addition, this applies the following fixes:
- The widged didn't work when the field was a `datetime` due to a typo
  in a JS conditional
- Replaced `Object.assign()` -> `_.assign()` to improve
  browser-compatibility

### CODE REVIEW

**Comment by yaniaular** [comment]
@veloutin @yvaucher @sbidoul @pedrobaeza I have migrated this module and I need your approval to merge it, thanks.

**Comment by pedrobaeza** [comment]
Hi @yaniaular, thanks for the migration, but I'm afraid this is not how it works. Reviewers can come to this PR because they are interested in the module itself, or you can review other PRs in the same repository or other similar ones and ask in exchange that they review yours. Current commit authors that you see can be general OCA maintainers like me and @sbidoul that don't know about the module itself, or even non active Odoo developers.

**Comment by yaniaular** [comment]
@pedrobaeza I think you get me wrong. I didn't mean I was going to merge this. I just wanted to tell you that I've migrated the module and that you may review it. I do not have permission to merge it anyway.

File: web_widget_datepicker_options/README.rst
Comment (simahawk) [inline web_widget_datepicker_options/README.rst]: png

File: web_widget_datepicker_options/README.rst
Comment (simahawk) [inline web_widget_datepicker_options/README.rst:29]: bug tracker and credits need update from template https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst

File: web_widget_datepicker_options/__init__.py
Comment (simahawk) [inline web_widget_datepicker_options/__init__.py]: remove encoding and use short license headers in all py files

File: web_widget_datepicker_options/__manifest__.py
Comment (simahawk) [inline web_widget_datepicker_options/__manifest__.py]: `'website': 'https://github.com/OCA/web',`

File: web_widget_datepicker_options/__manifest__.py
Comment (simahawk) [inline web_widget_datepicker_options/__manifest__.py]:
```
'name': 'Web widget datepicker options',
'summary': 'Enhance customization for datepicker widgets',
```

File: web_widget_datepicker_options/static/src/js/datepicker.js
Comment (simahawk) [inline web_widget_datepicker_options/static/src/js/datepicker.js]: define `var parent = this.getParent();` and replace all `this.__parentedParent`

File: web_widget_datepicker_options/view/qweb.xml
Comment (simahawk) [inline web_widget_datepicker_options/view/qweb.xml]: s/openerp/odoo
remove `<data />`

File: web_widget_datepicker_options/README.rst
Comment (simahawk) [inline web_widget_datepicker_options/README.rst]: Am I wrong or you should use JSON notation?
```xml
<field name="date" options='{"datepicker":{"yearRange": "c-100:c+0"}}'/>
```

**Review by simahawk — CHANGES_REQUESTED** [review]

File: web_widget_datepicker_options/__manifest__.py
Comment (MeetKD) [inline web_widget_datepicker_options/__manifest__.py]: Add only web module dependency.

File: web_widget_datepicker_options/README.rst
Comment (luisg123v) [reply web_widget_datepicker_options/README.rst]: @simahawk 
Both single and double quotes are possible, it keeps the same behavior as Odoo's native options.

**Comment by luisg123v** [comment]
@simahawk & @MeetKD 
We applied all your requested changes. Also we edited the PR's description, to make clearer what this is all about.

If it's not too much, could you review again, please?

**Review by simahawk — APPROVED** [review]
LGTM. You could squash last 2 commits before merge.

**Comment by luisg123v** [comment]
@simahawk 
Are you sure squashing is the way to go? Take into account the author differs for those commits.

**Comment by luisg123v** [comment]
@moylop260 
Could you approve this PR, please?

**Comment by moylop260** [comment]
Could you squash the Yanina's commits in one commit, please?


UPDATED:
I mean, the following commits
 - https://github.com/OCA/web/pull/924/commits/68d4bde228e6983c1cf9f6af9a09e89901623899
 - https://github.com/OCA/web/pull/924/commits/c00ebff968286d7bddd4ab4288a6ce603b870c21

**Comment by luisg123v** [comment]
@moylop260 
Done, commits were squashed as you requested

**Review by yajo — COMMENTED** [review]
I might be saying some stupidity, but [isn't this already what the v11 core widget does](https://github.com/odoo/odoo/blob/1039abfbdb1dcfd51e753874795721fc0d617485/addons/web/static/src/js/widgets/date_picker.js#L22-L57)?

**Comment by luisg123v** [comment]
@Yajo 
AFAIK, it's not the same. That method merge a parameter named `options`, with some computed values. However, that parameter is not the options passed to the field in a view, but some common values that Odoo passes to all widgets

If you wan to, e.g. limit a field, the following won't work:
```
<field name="post_date" options="{'daysOfWeekDisabled': [0, 6]}"/>
```
But, with this module, you could do:
```
<field name="post_date" options="{'datepicker': {'daysOfWeekDisabled': [0, 6]}}"/>
```

**Review by MeetKD — APPROVED** [review]

**Comment by pedrobaeza** [comment]
I think first 8 commits from same author can be squashed, as that commit history doesn't add anything useful. When done, I'll merge.

**Comment by yaniaular** [comment]
@pedrobaeza Do you mean the veloutin's commits?

**Comment by pedrobaeza** [comment]
Yeah, that's it

**Comment by luisg123v** [comment]
@pedrobaeza 
Done, all commits from the same author were squashed.

**Comment by yajo** [comment]
Fixing this upstream is 1 LOC, I did it here: https://github.com/odoo/odoo/pull/25044

Let's hope they merge it and we need no more this addon :blush:

**Pull request merged**
